### PR TITLE
Some clean up on open issues

### PR DESCRIPTION
**Pull request by dblodgett-usgs**

### CODE REVIEW

File: R/REST_helpers.R [hunk 1]
```diff
@@ -92,6 +92,14 @@ sbtools_PUT <- function(url, body, ..., session) {
 #' @keywords internal
 sbtools_DELETE <- function(url, ..., session) {
 	check_session(session)
+	
+	uid <- tryCatch(user_id(session = session),
```
Comment (dblodgett-usgs) [inline R/REST_helpers.R:96]: The first time you hist this, I am now caching the user_id in the pkg.env so it's free for every subsequent delete after the first one.

File: R/REST_helpers.R [hunk 2]
```diff
@@ -121,6 +129,11 @@ handle_errors <- function(x, url, method, types) {
 		if(x$status == 404) warning("Could not access sciencebase")
 		return(NULL)
 	}
+		
+	if(x$status_code == 403) {
+		warning("Sciencebase returned '403 Forbidden'")
+		return(NULL)
+	}
```
Comment (dblodgett-usgs) [inline R/REST_helpers.R:136]: This fixes the issue raised in #272

File: R/item_file_download.R [hunk 3]
```diff
@@ -68,10 +68,18 @@ item_file_download = function(sb_id, ..., names, destinations,
...
+			GET(url=flist[i,]$url, ..., 
+					write_disk(flist[i,]$dest, overwrite = overwrite_file), 
+					handle=session, timeout = httr::timeout(default_timeout()))
+		}, error = function(e) {
+			if(file.exists(flist[i,]$dest)) {
+				warning(paste(basename(flist[i,]$dest), "exists, and overwrite is false. Skipping."))
+			} else {
+				stop(paste("Error downloading", flist[i,]$dest, "Original error: \n", e))
+			}
+		})
```
Comment (dblodgett-usgs) [inline R/item_file_download.R:82]: The above takes care of #271

File: R/item_replace_files.R [hunk 4]
```diff
@@ -7,9 +7,6 @@
 #' Defaults to FALSE.
 #' 
 #' @description replaces existing files associated with an item with a new one. 
-#'   (Currently does not support multi-file uploads.) This function will not
-#'   append an existing collection of files. If that is desired, use
-#'   \code{\link{item_append_files}}
```
Comment (dblodgett-usgs) [inline R/item_replace_files.R:12]: Takes care of #267

File: R/session_logout.R [hunk 5]
```diff
@@ -16,5 +16,6 @@ session_logout <- function(..., session = current_session()) {
 		stop("Logout did not succeed", call. = FALSE)
 	}
 	pkg.env$session <- NULL
+	try(pkg.env$uid <- NULL, silent = TRUE)
```
Comment (dblodgett-usgs) [inline R/session_logout.R:19]: Clean out the cache

**Comment by dblodgett-usgs** [comment]
@lindsayplatt -- have time to scan this PR for me?

**Comment by jzwart** [comment]
@dblodgett-usgs I could probably do a review early next week

**Comment by dblodgett-usgs** [comment]
Thanks @jzwart !!

**Review by jzwart — APPROVED** [review]
Looks good to me @dblodgett-usgs , just a single comment

**Comment by dblodgett-usgs** [comment]
Thanks @jzwart !!

**Pull request merged**